plugin/lighthouse/handler.go

```diff
@@ -111,6 +111,20 @@ func (lh *Lighthouse) emptyResponse(state request.Request) (int, error) {
 	return dns.RcodeSuccess, nil
 }
 
+func (lh *Lighthouse) getClusterIpForSvc(pReq recordRequest) (ip string, found bool) {
+	localClusterID := lh.clusterStatus.LocalClusterID()
+
+	ip, found, isLocal := lh.serviceImports.GetIP(pReq.namespace, pReq.service, pReq.cluster, localClusterID, lh.clusterStatus.IsConnected,
+		lh.endpointsStatus.IsHealthy)
+
+	getLocal := isLocal || (pReq.cluster != "" && pReq.cluster == localClusterID)
+	if found && getLocal {
+		ip, found = lh.localServices.GetIp(pReq.service, pReq.namespace)
+	}
```

Comment:
Perhaps it would cleaner to incorporate `localServices` into `serviceImports.GetIP`. But we can consider that for a separate PR.

Reply:
ServiceImport and Service are using two different informers. I have an idea on how to make it work, but will revisit it later after 0.8.0 With the plan to use EndpointSlice instead of ServiceImport for clusterIP, it will become easier.

Reply:
`LocalServices` is an interface so we could pass it into `serviceimport.Map`.

Reply:
Aha. Can do it in a separate PR later.